liquidation-monitor/src/agent.js

```diff
@@ -0,0 +1,500 @@
+    const exitMarketString = 'event MarketExited(address cToken, address account)';
+
+    const exitMarketEvents = txEvent.filterLog(exitMarketString);
+    exitMarketEvents.forEach((exitEvent) => {
+      ts(`${exitEvent.args.account} exited from ${exitEvent.args.account} `);
+      data.newAccounts.push(exitEvent.args.account);
+    });
+    const borrowEvents = txEvent.filterLog(borrowString);
+    borrowEvents.forEach((borrowEvent) => {
+      ts(`${borrowEvent.args.borrower} borrowed ${borrowEvent.args.borrowAmount} `);
+      data.newAccounts.push(borrowEvent.args.borrower);
+    });
+    // #endregion
+
+    // Return zero findings
+    return [];
+  };
+}
+
+function provideHandleBlock(data) {
+  return async function handleBlock(blockEvent) {
+    ts(`Starting block ${blockEvent.blockNumber}`);
+    const findings = [];
+    const { comptrollerContract, oneInchContract } = data;
+
+    // #region Add new Accounts
+    ts(
+      `Currently tracking ${Object.keys(data.accounts).length
+      } accounts and adding ${data.newAccounts.length
+      } account(s) this block. ${data.totalNewAccounts
+      } total accounts added since start.`,
+    );
+    data.newAccounts.forEach((newAccount) => {
+      const account = newAccount.toLowerCase();
+      // Initialize account. New accounts will get updated in the block section
+      /* eslint-disable no-param-reassign */
+      data.accounts[account] = {};
+      /* eslint-enable no-param-reassign */
+    });
+    /* eslint-disable no-param-reassign */
+    data.totalNewAccounts += data.newAccounts.length;
+    data.newAccounts = [];
+    /* eslint-enable no-param-reassign */
+    // #endregion
+
+    // #region Update Balances on zero health accounts
+    const filteredAccounts = [];
+    Object.keys(data.accounts).forEach((currentAccount) => {
+      if (data.accounts[currentAccount].health == null
+        || data.accounts[currentAccount].health === 0) {
+        filteredAccounts.push(currentAccount);
+        // Zero account balances
+        /* eslint-disable no-param-reassign */
+        Object.keys(data.supply).forEach((currentToken) => {
+          data.supply[currentToken][currentAccount] = null;
+        });
+        Object.keys(data.borrow).forEach((currentToken) => {
+          data.borrow[currentToken][currentAccount] = null;
+          /* eslint-enable no-param-reassign */
+        });
+      }
+    });
+
+    // Grab the assets in first, and make sure they are initialized
+    ts('Getting assetsIn list for new accounts');
+    const tokenSet = new Set();
+    await Promise.all(filteredAccounts.map(async (currentAccount) => {
+      let assetsIn = await comptrollerContract.getAssetsIn(currentAccount);
+      assetsIn = assetsIn.map((asset) => asset.toLowerCase());
+      assetsIn.forEach((asset) => { tokenSet.add(asset); });
+      /* eslint-disable no-param-reassign */
+      data.accounts[currentAccount].assetsIn = assetsIn;
+      /* eslint-enable no-param-reassign */
```

Comment:
Assuming the eslint disable/enable comments are applied to the whole function.

This is just an alternative suggestion, potentially more readable?
```suggestion
      const assetsIn = await comptrollerContract.getAssetsIn(currentAccount);
      data.accounts[currentAccount].assetsIn = assetsIn.map((asset) => {
        const address = asset.toLowerCase();
        tokenSet.add(address);
        return address;
      });
```